cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/FileContentWriter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service.upload;
+
+import co.cask.cdap.common.io.Locations;
+import co.cask.cdap.data.stream.StreamDataFileConstants;
+import co.cask.cdap.data.stream.StreamDataFileWriter;
+import co.cask.cdap.data.stream.service.ConcurrentStreamWriter;
+import co.cask.cdap.data.stream.service.MutableStreamEvent;
+import co.cask.cdap.data.stream.service.MutableStreamEventData;
+import co.cask.cdap.data2.transaction.stream.StreamConfig;
+import com.google.common.collect.Maps;
+import com.google.common.io.Closeables;
+import org.apache.twill.filesystem.Location;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Implementation of {@link ContentWriter} that writes to stream file directly.
+ */
+final class FileContentWriter implements ContentWriter {
+
+  private final String accountId;
+  private final StreamConfig streamConfig;
+  private final ConcurrentStreamWriter streamWriter;
+  private final MutableStreamEventData streamEventData;
+  private final MutableStreamEvent streamEvent;
+  private final Location eventFile;
+  private final Location indexFile;
+  private final StreamDataFileWriter writer;
+  private long eventCount;
+
+  FileContentWriter(String accountId, StreamConfig streamConfig, ConcurrentStreamWriter streamWriter,
+                    Location directory, Map<String, String> headers) throws IOException {
+    this.accountId = accountId;
+    this.streamConfig = streamConfig;
+    this.streamWriter = streamWriter;
+    this.streamEventData = new MutableStreamEventData();
+    this.streamEvent = new MutableStreamEvent();
+
+    directory.mkdirs();
+    this.eventFile = directory.append(String.format("upload.%s.dat", streamConfig.getName()));
+    this.indexFile = directory.append(String.format("upload.%s.idx", streamConfig.getName()));
+
+    Map<String, String> properties = createStreamFileProperties(headers);
+    properties.put(StreamDataFileConstants.Property.Key.UNI_TIMESTAMP,
```

Comment:
No. The constant is called `UNI_TIMESTAMP`. It's a special property saying all events in the stream file has the same timestamp.